### PR TITLE
Add a status summary to the README.

### PR DESCRIPTION
**Pull request by sunfishcode**
This adds a brief blurb about Cretonne's current status, so that people looking at Cretonne have an idea of what to expect.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -23,6 +21,37 @@ machine code.
...
+
+Cretonne does not yet perform mitigations for Spectre or related security
+issues, though it may do so in the future. It does not currently make any
+security-relevant instruction timing guarantees. It has seen a fair amount
+of testing and fuzzing, although more work is needed before it would be
+ready for a production use case.
+
+Cretonne's APIs are not yet stable.
+
+Cretonne currently supports Rust 1.22.1 and later. We indent to always support
```
Comment (varkor) [inline README.rst]: Typo: intend.

File: README.rst [hunk 2]
```diff
@@ -23,6 +21,37 @@ machine code.
...
+
+The x86-64 backend is currently the most complete and stable; other
+architectures are in various stages of development. Cretonne currently supports
+the System V AMD64 ABI calling convention used on many platforms, but does not
+yet support the Windows x64 calling convention. The performance of code
+produced by Cretonne is not yet impressive, though we have plans to fix that.
+
+The core codegen creates have minimal dependencies, and do not require any host
+floating-point support. Support for `no_std` mode in the core codegen crates is
+[in development](https://github.com/Cretonne/cretonne/tree/no_std).
```
Comment (pepyakin) [inline README.rst]: Is this a correct link format?

File: README.rst [hunk 3]
```diff
@@ -23,6 +21,37 @@ machine code.
...
+functions, although it needs to be used within an external WebAssembly
+embedding to be part of a complete WebAssembly implementation.
+
+The x86-64 backend is currently the most complete and stable; other
+architectures are in various stages of development. Cretonne currently supports
+the System V AMD64 ABI calling convention used on many platforms, but does not
+yet support the Windows x64 calling convention. The performance of code
+produced by Cretonne is not yet impressive, though we have plans to fix that.
+
+The core codegen creates have minimal dependencies, and do not require any host
```
Comment (bjfish) [inline README.rst]: `The core codegen creates have minimal dependencies, and do not require` Is the grammar correct here? I'm having trouble understanding this sentence.

Comment (pepyakin) [reply README.rst]: I think there is a typo: creates → crates

Comment (sunfishcode) [reply README.rst]: @bjfish If I split it into two sentences, does that clarify it? "The core codegen creates have minimal dependencies. And, they do not require any host floating-point support."

File: README.rst (hunk 2 above)
Comment (sunfishcode) [reply README.rst]: Oops, I used markdown format, and this is a reST file. Thanks for spotting that! I'll fix it.

File: README.rst (hunk 3 above)
Comment (bjfish) [reply README.rst]: @sunfishcode I still don't understand. Is there a typo? Should it be: "The *code* codegen creates *has* minimal dependencies." or "The core codegen *crates* have minimal dependencies." Or is "creates" actually a noun in this context? If so what are "creates"?

Comment (sunfishcode) [reply README.rst]: Ah, I see it now. It's a typo. "creates" should be "crates". Thanks for spotting that!

**Comment by sunfishcode** [comment]
Thanks for the comments everyone! If there are any further questions about status, feel free to file new issues and/or ask on [gitter](https://gitter.im/Cretonne/Lobby).

**Pull request merged**